addons/stock/models/stock_inventory.py

```diff
@@ -179,6 +192,9 @@ def _get_inventory_lines_values(self):
         locations = self.env['stock.location'].search([('id', 'child_of', [self.location_id.id])])
         domain = ' location_id in %s'
         args = (tuple(locations.ids),)
+        vals = []
+        Product = products = self.env['product.product']
```

Comment:
By the way, if you do 'products |= categ_prods', you modify products but ALSO Products.